src/java/org/apache/cassandra/db/compaction/UnifiedCompactionContainer.java

```diff
@@ -100,6 +100,11 @@ public boolean isActive()
         return strategy.isActive;
     }
 
+    private boolean hasVector()
+    {
+        return factory.getRealm().metadata().hasVectorType();
```

Comment:
Because `metadata()` resolves a `TableMetadataRef`, what we are getting with this call is whether the _updated_ metadata has vector types. In other words, the `hasVector() != isVectorTable` test below cannot become true.

What we actually need is to store the `hasVector` state at construction time and compare that against `factory.getRealm().metadata().hasVectorType()`. This also means that the change to the `shouldReload` signature is not necessary.